game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/change/CheckGeneralBattleEndOld.java

```diff
@@ -4,6 +4,7 @@
 import games.strategy.triplea.delegate.ExecutionStack;
```

Comment:
The `Old` was to differentiate it from the non `Old` class.  Unfortunately, since they are part of the save game, we can't change their names :(

Reply:
This code is new since the last release (right?) -  sparingly, you can break prerelease save games. The latest release is where we need to absolutely need to guarantee compatibility unless we agree to do a major version release.  It's somewhat assumed that those that stumble upon a prerelease know what they are doing and compatibility is not 100% guaranteed.

Reply:
Yes, the CheckGeneralBattleEnd* classes were added in 2.3.21665 which I believe is a prerelease.  Also, I think I'm mistaken on the `Old` class.  It shouldn't be included in any save.  The `CheckGeneralBattleEnd` class, though, will be.

Reply:
Any objects then to removing the 'General' word from the class names?

IMO this is just as descriptive of a name `CheckBattleEnd` as is: `CheckGeneralBattleEnd`

Reply:
`General` stands for the "general phase" (that's what the rule books generally call it).  I would expect to add a `CheckFirstStrikeBattleEnd` and a `CheckAaBattleEnd` at some point.  So I'd like to keep `General` around.

Reply:
In this context, the word `General` IMO simply adds nothing.

Seeing these two would be fine by me:
- CheckFirstStrikeBattleEnd
- CheckBattleEnd

I'll agree to disagree with you here. When you've lost context on this code and come back to it in a few weeks or months, perhaps in the back of your mind, check if you still have the same opinion and if the word 'general' adds more information or is actually more noise.